### ISSUE TITLE
Proof of Concept Firebase A/B Testing Implementation 

### ISSUE THREAD

**Issue by ajsarkar28** [department-of-veterans-affairs/va-mobile-app#9679]
Objective: Determine the technical feasibility of integrating Firebase A/B Testing with the app using Remote Config. The focus is to understand the setup process, necessary changes to the app’s architecture, and any potential constraints.

AC
- Setup/Implement firebase website and library requirements to trigger an A/B response on a developer screen test and review analytics provided
- If test is implemented in Prod, contact QA to check that Prod is fine

**Comment by theodur** [comment]
The mobile app has everything in place for running A/B testing in Firebase. To start a new experiment or see existing experiments, you can visit the [A/B Tests tab](https://console.firebase.google.com/project/va-mobile-app/config/env/firebase/experiments) in the Remote Config section of the Firebase console. 

I created 2 experiments for this spike, 1 for iOS and 1 for Android, since each OS needs its own experiment. Experiments are focused around audiences and variants that are determined by feature toggle values. For the sake of this spike, I used the `testFeature` feature toggle for the experiments, since it isn't being used. I also used the audience of Staging users so that this would only impact staging, but we can select audiences from this [list](https://console.firebase.google.com/project/va-mobile-app/analytics/app/android:gov.va.mobileapp/audience/audiences~2Fhub%3Ffpn%3D226948484727). We also can create a new audience for whatever experiment we want based on user properties.

I had 2 variants for these experiments, 1 being the baseline (the `testFeature` feature toggle set to its default value of `false`) and variant A (`testFeature` being set to `true`). When `testFeature` is `true`, I displayed a message in the Developer screen. You can add up to 5 variants in an experiment. You can also set the weight distributions of the variants, so if you want a higher percentage of people to see variant A, you can set the percentage higher compared to the baseline. There are also more customizable feature for A/B tests, like setting an activation event, which can be a way of filtering which users are included in the experiment measurements. From Firebase:

> An event that activates the experiment for a user. All targeted experiment users will receive an experiment variant treatment from the start of the experiment, but only users who trigger the activation event will be included in experiment measurement.

One issue I ran into when testing was the remote config override being set in the app, which prevents the app from fetching feature toggle values from Firebase, and just uses the local remote config values. This seems to be caused by the waygate logic setting the override value to `true` when loading the waygate values. This would only impact testing A/B testing in the future, but wouldn't impact rolling out an experiment to prod, so as of now, the app is ready for A/B testing in production if we wanted to. We should still look into this to make sure the override value isn't being unnecessarily set to `true` to make testing on staging easier.


Other than adding the appropriate feature toggle and updating the code to use that feature toggle, and creating the experiment in Firebase, there's nothing else we'd need to do in order to run an A/B experiment.

**Comment by theodur** [comment]
Moved this to code review while it's being reviewed by @ajsarkar28 and @dumathane

**Comment by theodur** [comment]
After reviewing my finding above, Ameet suggested we run a trial A/A test in production, meaning a test where nothing has changed but confirms that traffic is flowing appropriately. For the A/A test, we decided to use `vama_rx_refill_success` as the primary goal of the experiment, and `vama_rx_request_confirm` as the activation event.

I created an experiment for both [iOS](https://console.firebase.google.com/project/va-mobile-app/config/experiment/results/4) and [Android](https://console.firebase.google.com/project/va-mobile-app/config/experiment/results/5). After about a week of running the experiment, the iOS experiment has 57,000 total users and the Android experiment had 21,000 users, which accurately correlates to the distribution of iOS/Android user who use the app. Users across each variant were spread out evenly, at 50% receiving the `Baseline` and 50% receiving `variant A`. This suggests the experiment distribution and tracking is working.
